packages/smooth_app/lib/helpers/picture_capture_helper.dart

```diff
@@ -1,39 +1,105 @@
 import 'package:smooth_app/database/product_query.dart';
-import 'package:smooth_app/generic_lib/loading_dialog.dart';
+import 'package:workmanager/workmanager.dart';
+
+@pragma(
+    'vm:entry-point') // Mandatory if the App is obfuscated or using Flutter 3.1+
+void callbackDispatcher() {
+  Workmanager().executeTask(
+    (String task, Map<String, dynamic>? inputData) async {
+      // make a counter with task as key as it is unique for each task
+      final int counter = inputData!['counter'] as int;
+      // if task is greate than 6 , that means it has been executed 7 times
+      if (counter > 6) {
+        // returns true to let platform know that the task is completed
+        final File file = File(inputData['imageUri'].toString());
+        file.delete();
+        return Future<bool>.value(true);
+      }
+      final List<Duration> duration = <Duration>[
+        const Duration(seconds: 30),
+        const Duration(minutes: 1),
+        const Duration(minutes: 30),
+        const Duration(hours: 1),
+        const Duration(hours: 6),
+        const Duration(days: 1),
+      ];
+      bool shouldRetry = false;
+      try {
+        final SendImage image = SendImage(
+          lang: ProductQuery.getLanguage(),
+          barcode: inputData['barcode'].toString(),
+          imageField:
+              ImageFieldExtension.getType(inputData['imageField'].toString()),
+          imageUri: Uri.parse(inputData['imageUri'].toString()),
+        );
+        final Status result = await OpenFoodAPIClient.addProductImage(
+          ProductQuery.getUser(),
+          image,
+        );
+        shouldRetry = result.error != null || result.status != 'status ok';
+      } catch (e) {
+        shouldRetry = true;
+      }
+      if (shouldRetry) {
+        inputData['counter'] = counter + 1;
+        await Workmanager().initialize(callbackDispatcher);
+        await Workmanager().registerOneOffTask(
+          task,
+          'ImageUploadWorker',
+          constraints: Constraints(
+            networkType: NetworkType.connected,
+          ),
+          inputData: inputData,
+          initialDelay: duration[counter],
+        );
+        return Future<bool>.error('Failed and it will try again');
+      } else {
+        // go to the file system and delete the file that was uploaded
+        final File file = File(inputData['imageUri'].toString());
+        file.delete();
+        return Future<bool>.value(true);
+      }
+    },
+  );
+}
 
 Future<bool> uploadCapturedPicture(
   BuildContext context, {
   required String barcode,
   required ImageField imageField,
   required Uri imageUri,
 }) async {
+  // ignore: unused_local_variable
   final AppLocalizations appLocalizations = AppLocalizations.of(context);
-  final SendImage image = SendImage(
-    lang: ProductQuery.getLanguage(),
-    barcode: barcode,
-    imageField: imageField,
-    imageUri: imageUri,
+  final Map<String, dynamic> inputData = <String, dynamic>{
```

Comment:
Following my comment, please wrap this content in a class, with serialize/deserialize operations